### PR TITLE
Added support to new checkbox plugin

### PR DESCRIPTION
**Pull request by rmattos**
Refactoring of _handleEvent method to support checkbox plugin callback on authentication event and added new tests to validate the code.

### CODE REVIEW

**Comment by brenohq** [comment]
Nice!

**Comment by Charca** [comment]
Thanks @rmattos! I'll test it this afternoon and will merge/deploy later today.

**Comment by Charca** [comment]
Hey @rmattos, sorry for the huge delay. I just found time to look at this today and get familiar with the checkbox plugin.

This definitely works, but I have some concerns about storing the `optinOrigin` as an attribute of the Bot instance. I think it belongs more to the Chat instance, as we'll use it to reply to that particular user only. It won't affect most users, but I can see some edge cases where trying to send a message to a different user than the one that has just logged in (using `bot.say()` and not `chat.shay()`) could cause problems.

Also, I don't think we need to store the `optinOrigin` at all, I believe we can solve this without potential side effects by making two small changes:

1. We'll need to refactor `sendAction` and `sendMessage` to accept either an object or a string as the `recipientId` param.
2. In `_handleEvent` we'll check if it's a checkbox event and we'll instantiate the Chat instance using an object with the proper `user_ref` format.

So basically:

```javascript
_handleEvent(type, event, data) {
  const recipient = (type === 'authentication' && !event.sender) ? { user_ref: event.optin.user_ref } : event.sender.id;
  const chat = new Chat(this, recipient);
  this.emit(type, event, chat, data);
}
```

And your `_createRecipient` method could simply be:

```javascript
_createRecipient(recipient) {
  return (typeof recipient === 'object') ? recipient : { id: recipient };
}
```

Does that make sense?

Let me know if you have any questions or if you want me to commit my changes (I'll need permissions to push to your branch) and take a closer look.

**Comment by rmattos** [comment]
Hi @Charca thanks for review and improvements . I sent an invite to you to allow the access to the branch, so you can merge your changes and after that i think you can finally finish the pull request! =)

**Comment by Charca** [comment]
Hey @rmattos, I've just updated your branch with my proposed changes, let me know if you want to take a look at it and give it a shot before I merge this in. Cheers!

**Comment by rmattos** [comment]
Hey @Charca it´s seems ok to me! tks
